ludwig/schema/model_config.py

```diff
@@ -511,7 +513,11 @@ def _set_hyperopt_defaults(self):
         if not self.hyperopt:
             return
 
-        scheduler = self.hyperopt.get("executor", {}).get("scheduler")
+        # Convert hyperopt config to hyperopt schema to populate with schema defaults
+        # This fills in missing splits, executor config, search_alg, etc.
+        self.hyperopt = HyperoptConfig.Schema().load(self.hyperopt).to_dict()
```

Comment:
Should be:

```
HyperoptConfig.from_dict(..).to_dict()
```

Otherwise we circumvent the class cache layer and leak memory.

Reply:
@tgaddair You mean `ModelConfig.from_dict`, right?